### PR TITLE
Ubuntu 20.04. CI/CD Pipeline as GitHub Action

### PR DESCRIPTION
**Pull request by udosson**
This PR solves #1678 and #1679 and provides the GHA workflow for the CI and the CD pipeline of Indy-Node.
Also, this PR contains an update of the `ubuntu-20.04-upgrade` feature branch with the most recent commits of `master`.
The PR contains:
- CI workflow based on `master` branch and Indy-Plenum
- CD workflow based on `master` branch and Indy-Plenum
- Fixed linting issues with newer version of `flake8` (flake8==3.8.4) and `python 3.8`
- updated the version of Indy-SDK to `python3-indy==1.15.0-dev-1625`, the same version as used in Indy-Plenum which contains the freeze leger support.
- adjusted `runner.py` to be compatible with the newer version of `pytest` (pytest>=6.2.2)
- `install_requires` in `setup.py` is based on `indy-plenum==1.13.0.dev135` which was built and published from the `ubuntu-20.04-upgrade` branch of Indy-Plenum in this [GHA run](https://github.com/udosson/indy-node/actions/runs/1230157953)
- copied the uploading structure of the debian file to jfrog artifactory from `master` branch

A successful run of the workflow can be found at: https://github.com/udosson/indy-plenum/actions/runs/1186783441

Thanks to @WadeBarnes for all your support!

### CODE REVIEW

**Comment by udosson** [comment]
@WadeBarnes, thanks for your review.
I squashed all commits from `6f2d3b6` on into one commit.
image

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,440 @@
+name: indy-node-build
+on: [ push, pull_request, workflow_dispatch ]
```
Comment (WadeBarnes) [inline .github/workflows/build.yaml:2]: @udosson, I think the following `on` section from `indy-plenum` should be used here:
https://github.com/hyperledger/indy-plenum/blob/4547fc48d113e854f49c38569a26d645d3f5608e/.github/workflows/build.yaml#L2-L11

Comment (udosson) [reply .github/workflows/build.yaml]: agreed!

**Review by Echsecutor — COMMENTED** [review]
\n @ EOF might be nice ;)

**Review by WadeBarnes — APPROVED** [review]

**Pull request merged**